helix-core/src/main/java/org/apache/helix/messaging/handling/HelixTaskExecutor.java

```diff
@@ -1288,6 +1308,19 @@ private void removeMessageFromZK(HelixDataAccessor accessor, Message message,
     }
   }
 
+  private void sendNopMessage(HelixDataAccessor accessor, String instanceName) {
```

Comment:
This function will trigger another event and retry the message, right? Why in the PR it said the message will not be automatically retried?

Reply:
Oh yeah, good catch. I changed the design since during the test I find that would be a very bad way if we leave the message there and pray for some new messages comming.
I will modify the description.